### PR TITLE
[google-apps-script] add url methods to SpreadsheetApp.RichTextValue and SpreadsheetApp.RichTextValueBuilder

### PR DESCRIPTION
**Pull request by therealchjones**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developers.google.com/apps-script/reference/spreadsheet/rich-text-value & https://developers.google.com/apps-script/reference/spreadsheet/rich-text-value-builder
- [ ***unknown*** ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Fixes #54271  and adds tests for SpreadsheetApp.RichTextValue and SpreadsheetApp.RichTextValueBuilder

### CODE REVIEW

File: types/google-apps-script/google-apps-script-tests.ts [hunk 1]
```diff
@@ -142,6 +142,35 @@ const request: GoogleAppsScript.Data_Studio.Request<any> = {
...
+// Spreadsheet Rich Text Value & Builder
+// starting with sample code from
+// https://developers.google.com/apps-script/reference/spreadsheet/rich-text-value-builder
+const richTextStyle = SpreadsheetApp.newTextStyle()
+    .setUnderline(false)
+    .build();
+const richTextValue = SpreadsheetApp.newRichTextValue()
+    .setText("foo no baz")
+    .setLinkUrl(0, 3, "https://bar.foo")
+    .setLinkUrl(7, 10, "https://abc.xyz")
```
Comment (mtgto) [inline types/google-apps-script/google-apps-script-tests.ts:154]: It is prefer to use null for 3rd argument.

**Review by mtgto — APPROVED** [review]
LGTM. Thanks for your contribution 👍

File: types/google-apps-script/google-apps-script-tests.ts (hunk 1 above)
Comment (therealchjones) [reply types/google-apps-script/google-apps-script-tests.ts:154]: Sorry, just for curiosity, which 3rd argument? These lines were copied from the Google documentation example. setLinkUrl(I,j,null) removes an existing link.

**Comment by therealchjones** [comment]
Ready to merge

Thank you for the opportunity!

**Comment by therealchjones** [comment]
Fixes #54272

**Pull request merged**